src/components/LearnerActivityTable/index.jsx

```diff
@@ -1,152 +1,200 @@
+  const currentPageFromUrl = pageFromUrl - 1; // Convert to zero-based for DataTable
+
+  const apiFieldsForColumnAccessor = {
+    userEmail: { key: 'user_email' },
+    courseTitle: { key: 'course_title' },
+    courseListPrice: { key: 'course_list_price' },
+    courseStartDate: { key: 'course_start_date' },
+    courseEndDate: { key: 'course_end_date' },
+    passedDate: { key: 'passed_date' },
+    currentGrade: { key: 'current_grade' },
+    progressStatus: { key: 'progress_status' },
+    lastActivityDate: { key: 'last_activity_date' },
+  };
 
-  formatTableData = enrollments => enrollments.map(enrollment => ({
-    ...enrollment,
-    user_email: <span data-hj-suppress>{enrollment.user_email}</span>,
-    last_activity_date: i18nFormatTimestamp({ intl: this.props.intl, timestamp: enrollment.last_activity_date }),
-    course_start_date: i18nFormatTimestamp({ intl: this.props.intl, timestamp: enrollment.course_start_date }),
-    course_end_date: i18nFormatTimestamp({ intl: this.props.intl, timestamp: enrollment.course_end_date }),
-    enrollment_date: i18nFormatTimestamp({
-      intl: this.props.intl, timestamp: enrollment.enrollment_date,
-    }),
-    passed_date: i18nFormatPassedTimestamp({ intl: this.props.intl, timestamp: enrollment.passed_date }),
-    user_account_creation_date: i18nFormatTimestamp({
-      intl: this.props.intl, timestamp: enrollment.user_account_creation_date,
-    }),
-    progress_status: i18nFormatProgressStatus({ intl: this.props.intl, progressStatus: enrollment.progress_status }),
-    course_list_price: enrollment.course_list_price ? `$${enrollment.course_list_price}` : '',
-    current_grade: formatPercentage({ decimal: enrollment.current_grade }),
-  }));
-
-  render() {
-    const { activity, id } = this.props;
-    return (
-      <TableContainer
-        id={id}
-        className={id}
-        key={id}
-        fetchMethod={(enterpriseId, options) => EnterpriseDataApiService.fetchCourseEnrollments(
-          enterpriseId,
-          {
-            learnerActivity: activity,
-            ...options,
-          },
-        )}
-        columns={this.getTableColumns()}
-        formatData={this.formatTableData}
-        tableSortable
-      />
-    );
+  const {
+    isLoading,
+    courseEnrollments,
+    fetchCourseEnrollments,
+    hasData,
+  } = useCourseEnrollments(enterpriseId, id, apiFieldsForColumnAccessor);
+
+  // Update context when data status changes
+  useEffect(() => {
+    setTableHasData(id, hasData);
+  }, [id, hasData]);
+
+  // Wrap fetchCourseEnrollments to update the URL when pagination changes
+  const fetchTableData = useCallback((tableState) => {
+    const newPageForUrl = tableState.pageIndex + 1; // Convert zero-based index to one-based for URL
+
+    updateUrlWithPageNumber(newPageForUrl, location, navigate);
+
+    return fetchCourseEnrollments(tableState);
+  }, [fetchCourseEnrollments]);
```

Comment:
[curious] Could this be collocated within `useCouresEnrollments`? The learner allocation table uses a similarly constructed [useBudgetRedemptions](https://github.com/openedx/frontend-app-admin-portal/blob/master/src/components/learner-credit-management/data/hooks/useBudgetRedemptions.js#L49) hook to populate data and retrieve access to the table metadata in the top level hook. 

This pattern allows us to parse fields from the datatable (filters and sortBy) directly within the hook and use it as needed (track event metadata, logging errors/infos, possibly url generation).

Reply:
I can consider this a non-blocking nit but something to hopefully strive for as an in future iterations down the road.